Microsoft.HealthVault.Fhir/Transformers/PractitionerToHealthVault.cs

```diff
@@ -0,0 +1,136 @@
+                Name = new ItemTypes.Name()
+                {
+                    Last = practitionerName.Family,
+                    Suffix = practitionerName.Suffix.Any() ? new ItemTypes.CodableValue(practitionerName.Suffix.First()) : null,
+                    Title = practitionerName.Prefix.Any() ? new ItemTypes.CodableValue(practitionerName.Prefix.First()) : null,
+                    First = practitionerName.Given.FirstOrDefault() ?? string.Empty,
+                    Middle = practitionerName.Given.ElementAtOrDefault(1) ?? string.Empty,
+                },
+                ContactInformation = new ContactInfo()
+            };            
+            if (!string.IsNullOrEmpty(practitionerName.Text))
+            {
+                person.Name.Full = practitionerName.Text;
+            }
+                              
+
+            //address
+            if (!fhirPractitioner.Address.IsNullOrEmpty())
+            {
+                foreach (var address in fhirPractitioner.Address)
+                {
+                    var hvAddress = new ItemTypes.Address();
+
+                    foreach (var line in address.Line)
+                    {
+                        hvAddress.Street.Add(line);
+                    }
+                    hvAddress.City = address.City;
+                    hvAddress.State = address.State;
+                    hvAddress.County = address.District;
+                    hvAddress.PostalCode = address.PostalCode;
+                    if (!string.IsNullOrEmpty(address.Country))
+                    {
+                        hvAddress.Country = address.Country;
+                    }
+                        
+
+                    hvAddress.Description = address.Text;
+                    hvAddress.IsPrimary = address.GetBoolExtension(HealthVaultExtensions.IsPrimary);
+
+                    person.ContactInformation.Address.Add(hvAddress);
+                }
+            }
+
+            //telecom
+            if (!fhirPractitioner.Telecom.IsNullOrEmpty())
+            {
+                foreach (var contactPoint in fhirPractitioner.Telecom)
+                {
+                    switch (contactPoint.System)
+                    {
+                        case ContactPoint.ContactPointSystem.Email:
+                            person.ContactInformation.Email.Add(ConvertContactPointToEmail(contactPoint));
+                            break;
+                        case ContactPoint.ContactPointSystem.Phone:
+                            person.ContactInformation.Phone.Add(ConvertContactPointToPhone(contactPoint));
+                            break;
+                    }
+                }
+            }
+
+            //qualification               
+            if (!fhirPractitioner.Qualification.IsNullOrEmpty())
+            {
+                var firstQualification = fhirPractitioner.Qualification.FirstOrDefault(); //Let's take just the first one
```

Comment:
We've already checked that this shouldn't be empty, so we can just do a `First()`.

Reply:
Fixed; Changed to First()